test/entity.ts

```diff
@@ -1879,6 +1903,7 @@ describe('entity', () => {
       const query = ds
         .createQuery('Kind1')
         .filter('name', 'John')
+        .filter('__key__', 'IN', [new entity.Key({path: ['Kind1', 'key1']})])
```

Comment:
We should make a separate test case to test this functionality. Especially since the new tests added will not include this filter so can't be used against this proto.